### ISSUE TITLE
Suggestion : Use self-training instead of pre-training

### ISSUE THREAD

**Issue by pfeatherstone** [ultralytics/yolov5#477]
There has been some work on self-training applied to object detection models which has incurred huge success.
**https://github.com/tensorflow/tpu/tree/master/models/official/detection/projects/self_training**
**https://arxiv.org/pdf/2006.06882v1.pdf**
Looks like it uses the same concepts as noisy-student which, in the classification task, blitzed the Imagenet leader-board.
The authors claim up to +2AP. So might be worth putting in the pipeline for the future.

**Comment by Lornatang** [comment]
@pfeatherstone Yes, when the dataset is fully labeled, it is a good process to start training from the beginning!

**Comment by glenn-jocher** [comment]
@pfeatherstone could you summarize the steps you had in mind here? When training on COCO we don't use any pre-training.

**Comment by pfeatherstone** [comment]
I know diddly squat about this approach, only that the paper is claiming substantial benefits when using it. If it’s anything like noisy student, which I think it is, it uses a teacher network and a lot of unlabelled data.

**Comment by pfeatherstone** [comment]
Again, COCO has so much labelled data that the performance gains are probably minimal. My gut feeling is that these techniques stand out when you have a small custom LABELLED dataset, but a large unlabelled dataset.

**Comment by glenn-jocher** [comment]
@pfeatherstone right. The global wheat competition leaderboard used yolov5x with pseudolabel training of the test set as a second stage after training on the training set.

You can autolabel any folder of images using `python detect.py --save-txt`

**Comment by pfeatherstone** [comment]
I guess the point is that training methods are just as important if not much more important than model architectures. For example, the yolov3-spp model is a lot more accurate using the training loop from the ultralytics repo than the darknet repo. So it makes you wonder if the accuracy of the yolov5 models is attributed to the training loop rather than the model architectures. As an experiment, it would be interesting to see what MAP scores you get with yolov4 trained using the ultralytics repo. If the results show that yolov4 by ultralytics are much improved then it looks to me like efforts should be focussed on training methods rather than model architectures. Then again, I’m not an expert in this field.

**Comment by glenn-jocher** [comment]
@pfeatherstone yes this sounds about right to me. The darknet folks say that architecture is more important, but I would disagree, as YOLOv3-SPP as originally published by pjreddie scored 36AP I believe, whereas the same exact model (same anchors too), with zero changes is scoring about 46AP when trained with the YOLOv5 repo. YOLOv5l, which is comparable in size, but with a number of architecture changes, is scoring about 47. So 10/11, or about 90% of the mAP increase would be simply due to training improvements.
